WalletWasabi/Wallets/Wallet.cs

```diff
@@ -182,8 +182,8 @@ public List<TransactionSummary> BuildHistorySummary(bool sortForUi = false)
 			}
 			else
 			{
-				var unconfTransactionChainOfCoin = UnconfirmedTransactionChainProvider.GetUnconfirmedTransactionChain(coin.TransactionId) ?? [];
-				var effectiveFeeRate = FeeHelpers.CalculateEffectiveFeeRateOfUnconfirmedChain(unconfTransactionChainOfCoin);
+				var unconfTransactionChainOfCoin = UnconfirmedTransactionChainProvider.GetUnconfirmedTransactionChain(coin.TransactionId);
+				var effectiveFeeRate = unconfTransactionChainOfCoin is not null ? new FeeRate((decimal)unconfTransactionChainOfCoin.effectiveFeePerVsize) : null;
```

Comment:
Ok, this is not something introduced by this PR but here we make it worse, do not return `null` and do not pass `null` ar two basic rules that are being broken.

Reply:
I changed this to be a standard `TryGet`, so return `bool`.
But then not sure what you are suggesting to do?

If we can't have an `effectiveFeeRate` because we don't know it, don't it make sense to use `null`? Because the only alternatives I see are:
- Use `FeeRate(0)`, which makes absolutely no sense to me
- `throw`, which doesn't really make sense either because there is no problem, we simply have to wait, it's expected by design that the history will be built without this information that will come later.

Reply:
> If we can't have an effectiveFeeRate because we don't know it, don't it make sense to use null?

If you don't know the `effectiveFeeRate` then you cannot do anything with that. Sadly, the UI code uses a nullable field instead of designing the `TransactionSummary` (and others) properly. This is important because where there is a `null` it is a matter of time until you get a NRE. 

We don't need to go too far to see how this is bad. Look at this code. If you pass `null` then the `TransactionSummary::FeeRate()` can in some cases return null, this is use to create the `TransactionModel` and assign the `FeeRate` property which is then use to assign the `CoinJoinDetailsViewModel::FeeRate`  and finally that is used like this: `FeeRateVisible = FeeRate != FeeRate.Zero;` so, the fee rate is visible when `FeeRate` is not zero AND when it is `null`. So, in the xaml you have:

```
      <PreviewItem IsVisible="{Binding FeeRateVisible}"
                     Icon="{StaticResource paper_cash_regular}"
                     Label="Fee Rate"
                     CopyableContent="{Binding FeeRate.SatoshiPerByte}">
```
Boom! NRE.

Please, consider the use of `null`in the same way you consider the use of `goto` statuements.

Reply:
I fixed the NRE, I didn't touch Nullability. 

I understand what you say but we have a field `FeeRate` sometimes we can have the information sometimes we can't. Does it mean that not having it is a problem? No, we just don't have it. So throwing is not an option. If we don't have it should we consider it to be 0? No doing that is stupid, so I can't use `FeeRate(0)`. 


So now I have absolutely no idea what you want me to do.

Reply:
I am not saying that you have to change it, I am highlighting a problem. It is okay to fix it but that's not my point because every time you add a `null` you will end with a NRE so, the solution is to not use nulls. This is not a problem in your PR but a design problem in the `TransactionSummary` , and it is the same everywhere.

You have improved the code quality and fixed a bug, that's enough. But I want to let clear how that sucks.